core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -240,31 +278,37 @@ object ZkUtils extends Logging {
   /**
    *  make sure a persistent path exists in ZK. Create the path if not exist.
    */
-  def makeSurePersistentPathExists(client: ZkClient, path: String) {
+  def makeSurePersistentPathExists(client: ZkClient, path: String, acls: List[ACL] = DefaultAcls) {
+    //Consumer path is kept open as different consumers will write under this node.
+    val acl = if(path == null || path.isEmpty || path.equals(ConsumersPath)) {
+      import scala.collection.JavaConversions._
+      ZooDefs.Ids.OPEN_ACL_UNSAFE.toList
+    } else acls
```

Comment:
I apologise if this has already been explained, but why do we need to do this only in `makeSurePersistentPathExists` and not other `create*Path` methods?